pkg/agent/flowexporter/connections/l7_listener.go

```diff
@@ -0,0 +1,223 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connections
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/netip"
+	"os"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/config"
+	"antrea.io/antrea/pkg/agent/flowexporter"
+
+	// "antrea.io/antrea/pkg/agent/util"
+	k8sutil "antrea.io/antrea/pkg/util/k8s"
+	"antrea.io/antrea/pkg/util/podstore"
+)
+
+type PodL7VisibilityAttrGetter interface {
+	IsL7VisibilityRequested(podNN string, ingress bool) bool
+}
+
+// L7ProtocolFields holds layer 7 protocols supported
+type L7ProtocolFields struct {
+	http []map[int32]*Http
+}
+
+// Http holds the L7 HTTP flow JSON values.
+type Http struct {
+	Hostname      string `json:"hostname"`
+	URL           string `json:"url"`
+	UserAgent     string `json:"http_user_agent"`
+	ContentType   string `json:"http_content_type"`
+	Method        string `json:"http_method"`
+	Protocol      string `json:"protocol"`
+	Status        int32  `json:"status"`
+	ContentLength int32  `json:"length"`
+}
+
+// JsonToEvent holds Suricata event JSON values.
+// See https://docs.suricata.io/en/latest/output/eve/eve-json-format.html?highlight=HTTP%20event#event-types
+type JsonToEvent struct {
+	Timestamp   string     `json:"timestamp"`
+	FlowID      int64      `json:"flow_id"`
+	InInterface string     `json:"in_iface"`
+	EventType   string     `json:"event_type"`
+	VLAN        []int32    `json:"vlan"`
+	SrcIP       netip.Addr `json:"src_ip"`
+	SrcPort     int32      `json:"src_port"`
+	DestIP      netip.Addr `json:"dest_ip"`
+	DestPort    int32      `json:"dest_port"`
+	Proto       string     `json:"proto"`
+	TxID        int32      `json:"tx_id"`
+	HTTP        *Http      `json:"http"`
+}
+
+type L7Listener struct {
+	l7Events                  map[flowexporter.ConnectionKey]L7ProtocolFields
+	l7mut                     sync.Mutex
+	suricataEventSocketPath   string
+	podL7VisibilityAttrGetter PodL7VisibilityAttrGetter
+	podStore                  podstore.Interface
+}
+
+func NewL7Listener(
+	podL7VisibilityAttrGetterFunc PodL7VisibilityAttrGetter,
```

Comment:
```suggestion
	podL7VisibilityAttrGetter PodL7VisibilityAttrGetter,
```
it's an interface, not a function.